File: CS/ExportToXlsx/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Diagnostics;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraPrinting;
using System.IO;
using DevExpress.XtraGrid;

namespace ExportToXlsx
{
    public partial class Form1 : Form
    {
        ExportHelper eh;
        public Form1()
        {
            InitializeComponent();
            BindingList<Parent> list = new BindingList<Parent>();
            list.Add(new Parent(0, "Parent1"));
            list[0].Children.Add(new Child(Guid.NewGuid().ToString(), "Data"));
            list[0].Children.Add(new Child(Guid.NewGuid().ToString(), "Data1"));
            list.Add(new Parent(1, "Parent2"));
            list.Add(new Parent(2, "Parent3"));
            list[2].Children.Add(new Child(Guid.NewGuid().ToString(), "Data2"));
            list[2].Children.Add(new Child(Guid.NewGuid().ToString(), "Data3"));
            gridControl1.DataSource = list;
            eh = new ExportHelper(gridControl1);
        }

        private void simpleButton1_Click(object sender, EventArgs e)
        {
            eh.ExportToXlsx("test.xlsx");
            Process.Start("test.xlsx");
        }

        private void cb_MasterDetail_CheckedChanged(object sender, EventArgs e)
        {
            if (cb_MasterDetail.Checked)
                cb_MasterOnly.Checked = false;
            eh.MasterAndDetail = cb_MasterDetail.Checked;
        }

        private void cb_MasterOnly_CheckedChanged(object sender, EventArgs e)
        {
            if (cb_MasterOnly.Checked)
                cb_MasterDetail.Checked = false;
            eh.MasterOnly = cb_MasterOnly.Checked;
        }

        private void cb_NonBlank_CheckedChanged(object sender, EventArgs e)
        {
            eh.NonBlank = cb_NonBlank.Checked;
        }

    }
    public class Parent
    {
        int _Id;
        public int Id
        {
            get { return _Id; }
            set { _Id = value; }
        }

        string _Name;
        public string Name
        {
            get { return _Name; }
            set { _Name = value; }
        }

        BindingList<Child> _Children;
        public BindingList<Child> Children
        { get { return _Children; } }

        public Parent(int id, string name)
        {
            _Id = id;
            _Name = name;
            _Children = new BindingList<Child>();
        }
    }
    public class Child
    {
        public Child(string iD, string data)
        {
            _ID = iD;
            _Data = data;
        }

        string _ID;
        public string ID
        {
            get { return _ID; }
            set { _ID = value; }
        }

        string _Data;
        public string Data
        {
            get { return _Data; }
            set { _Data = value; }
        }
    }
}

File: VB/ExportToXlsx/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports System.Diagnostics
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraPrinting
Imports System.IO
Imports DevExpress.XtraGrid

Namespace ExportToXlsx
	Partial Public Class Form1
		Inherits Form
		Private eh As ExportHelper
		Public Sub New()
			InitializeComponent()
			Dim list As New BindingList(Of Parent)()
			list.Add(New Parent(0, "Parent1"))
			list(0).Children.Add(New Child(Guid.NewGuid().ToString(), "Data"))
			list(0).Children.Add(New Child(Guid.NewGuid().ToString(), "Data1"))
			list.Add(New Parent(1, "Parent2"))
			list.Add(New Parent(2, "Parent3"))
			list(2).Children.Add(New Child(Guid.NewGuid().ToString(), "Data2"))
			list(2).Children.Add(New Child(Guid.NewGuid().ToString(), "Data3"))
			gridControl1.DataSource = list
			eh = New ExportHelper(gridControl1)
		End Sub

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			eh.ExportToXlsx("test.xlsx")
			Process.Start("test.xlsx")
		End Sub

		Private Sub cb_MasterDetail_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cb_MasterDetail.CheckedChanged
			If cb_MasterDetail.Checked Then
				cb_MasterOnly.Checked = False
			End If
			eh.MasterAndDetail = cb_MasterDetail.Checked
		End Sub

		Private Sub cb_MasterOnly_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cb_MasterOnly.CheckedChanged
			If cb_MasterOnly.Checked Then
				cb_MasterDetail.Checked = False
			End If
			eh.MasterOnly = cb_MasterOnly.Checked
		End Sub

		Private Sub cb_NonBlank_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cb_NonBlank.CheckedChanged
			eh.NonBlank = cb_NonBlank.Checked
		End Sub

	End Class
	Public Class Parent
		Private _Id As Integer
		Public Property Id() As Integer
			Get
				Return _Id
			End Get
			Set(ByVal value As Integer)
				_Id = value
			End Set
		End Property

		Private _Name As String
		Public Property Name() As String
			Get
				Return _Name
			End Get
			Set(ByVal value As String)
				_Name = value
			End Set
		End Property

		Private _Children As BindingList(Of Child)
		Public ReadOnly Property Children() As BindingList(Of Child)
			Get
				Return _Children
			End Get
		End Property

		Public Sub New(ByVal id As Integer, ByVal name As String)
			_Id = id
			_Name = name
			_Children = New BindingList(Of Child)()
		End Sub
	End Class
	Public Class Child
		Public Sub New(ByVal iD As String, ByVal data As String)
			_ID = iD
			_Data = data
		End Sub

		Private _ID As String
		Public Property ID() As String
			Get
				Return _ID
			End Get
			Set(ByVal value As String)
				_ID = value
			End Set
		End Property

		Private _Data As String
		Public Property Data() As String
			Get
				Return _Data
			End Get
			Set(ByVal value As String)
				_Data = value
			End Set
		End Property
	End Class
End Namespace
